### PR TITLE
Make aliases created in compute() transformation visible for following transforms/query options

### PR DESCRIPTION
**Pull request by kosinsky**
### Issues

*This pull request fixes issue #1368 * also adds a few tests for ApplyBinder (it wasn't covered in original implementation)

### Description

Collecting aliases generated in compute() transformation to make them visible for following transformations and/or query options ($orderby, $filter)

### Checklist (Uncheck if it is not completed)

- [x] *Test cases added*
- [x] *Build and test with one-click build and test script passed*

### CODE REVIEW

File: test/FunctionalTests/Microsoft.OData.Core.Tests/UriParser/Parsers/UriQueryExpressionParserTests.cs [hunk 1]
```diff
@@ -653,6 +653,30 @@ public void ParseApplyWithNestedFunctionAggregation()
...
 
+        [Fact]
+        public void ParseApplyWithComputeWithMathematicalOperations()
+        {
+            string compute = "compute(Prop1 mul Prop2 as Product,Prop1 div Prop2 as Ratio,Prop2 mod Prop2 as Remainder)";
+            ComputeToken token = this.testSubject.ParseApply(compute).First() as ComputeToken;
+            token.Kind.ShouldBeEquivalentTo(QueryTokenKind.Compute);
+            List<ComputeExpressionToken> tokens = token.Expressions.ToList();
+            tokens.Count.Should().Be(3);
+            tokens[0].Kind.ShouldBeEquivalentTo(QueryTokenKind.ComputeExpression);
```
Comment (xuzhg) [inline test/FunctionalTests/Microsoft.OData.Core.Tests/UriParser/Parsers/UriQueryExpressionParserTests.cs]: Maybe it's more clear to create a private method to verify # 1, # 2, # 3 separately.

File: test/FunctionalTests/Microsoft.OData.Core.Tests/ScenarioTests/UriParser/FilterAndOrderByFunctionalTests.cs [hunk 2]
```diff
@@ -1163,6 +1179,58 @@ public void AggregatedPropertiesTreatedAsOpenPropertyInOrderBy()
...
+        }
+
+        [Fact]
+        public void ReferenceComputeAliasCreatedBeforeAggrageteThrows()
+        {
+            var odataQueryOptionParser = new ODataQueryOptionParser(HardCodedTestModel.TestModel,
+            HardCodedTestModel.GetPersonType(), HardCodedTestModel.GetPeopleSet(),
+            new Dictionary<string, string>()
+            {
+                            {"$orderby", "DoubleFavorite asc"},
```
Comment (xuzhg) [inline test/FunctionalTests/Microsoft.OData.Core.Tests/ScenarioTests/UriParser/FilterAndOrderByFunctionalTests.cs]: make the codes alignment

File: test/FunctionalTests/Microsoft.OData.Core.Tests/ScenarioTests/UriParser/FilterAndOrderByFunctionalTests.cs [hunk 3]
```diff
@@ -1163,6 +1179,58 @@ public void AggregatedPropertiesTreatedAsOpenPropertyInOrderBy()
...
+                            {"$apply", "compute(FavoriteNumber mul 2 as DoubleFavorite)/aggregate(DoubleFavorite with sum as Total)"}
+            });
+            Action parseAction = () => { odataQueryOptionParser.ParseApply(); odataQueryOptionParser.ParseOrderBy(); };
```
Comment (xuzhg) [inline test/FunctionalTests/Microsoft.OData.Core.Tests/ScenarioTests/UriParser/FilterAndOrderByFunctionalTests.cs]: Is "ParseOrderBy()" necessary in this test case?

Comment (kosinsky) [reply test/FunctionalTests/Microsoft.OData.Core.Tests/ScenarioTests/UriParser/FilterAndOrderByFunctionalTests.cs]: Yes. ParseApply() will work without problems and hide DoubleFavorite from ParseOrderBy() that should throw.

File: test/FunctionalTests/Microsoft.OData.Core.Tests/UriParser/Parsers/UriQueryExpressionParserTests.cs (hunk 1 above)
Comment (kosinsky) [reply test/FunctionalTests/Microsoft.OData.Core.Tests/UriParser/Parsers/UriQueryExpressionParserTests.cs]: We have test ParseComputeWithMathematicalOperations that checks $compute for that case. I'm adding the same for compute()

**Pull request merged**